nupic/research/temporal_memory.py

```diff
@@ -512,15 +597,17 @@ def adaptSegment(self, segment, activeSynapses, connections):
     @param segment        (int)         Segment index
     @param activeSynapses (set)         Indices of active synapses
     @param connections    (Connections) Connectivity of layer
+    @param permanenceIncrement (float)  Amount to increment active synapses
+    @param permanenceDecrement (float)  Amount to decrement inactive synapses
     """
```

Comment:
Could this also exist as an alternative with a table lookup index? A table of user and predefined permanence update callback functions. Allowing for functions to play with the permanence update in a non-linear way. At some point my reasoning suggests that dendritic distance from soma may smooth out some noise, Assuming potential, random limited-length segments upon synapse creation. Something to try out in community research?